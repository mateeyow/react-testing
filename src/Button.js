import React from 'react';

const Button = ({ variant = 'primary', children, onClick }) => (
  <button className={`button button-${variant}`} onClick={onClick}>
    {children}
  </button>
);

export default Button;
