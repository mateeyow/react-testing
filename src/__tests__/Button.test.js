import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../Button';

describe('<Button /> tests', () => {
  it('should render as default button', () => {
    const { container } = render(<Button />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a primary button', () => {
    const variant = 'primary';
    const { container } = render(<Button variant={variant}></Button>);

    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass(`button-${variant}`);
  });

  it('should render a secondary button', () => {
    const variant = 'secondary';
    const { container } = render(<Button variant={variant}></Button>);

    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass(`button-${variant}`);
  });

  it('should render the button with some text', () => {
    const text = 'I am a button, I can be clicked';

    const { getByText, container } = render(<Button>{text}</Button>);

    expect(container).toMatchSnapshot();
    expect(getByText(text)).toBeInTheDocument();
  });

  it('should call the click callback handler', () => {
    const content = 'Hello, world!'
    const onClick = jest.fn()
    const { getByText } = render(<Button onClick={onClick}>{content}</Button>)

    fireEvent.click(getByText(content))

    expect(onClick).toHaveBeenCalledTimes(1)
  })
});
