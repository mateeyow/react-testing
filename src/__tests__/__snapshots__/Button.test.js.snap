// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should render a primary button 1`] = `
<div>
  <button
    class="button button-primary"
  />
</div>
`;

exports[`<Button /> tests should render a secondary button 1`] = `
<div>
  <button
    class="button button-secondary"
  />
</div>
`;

exports[`<Button /> tests should render as default button 1`] = `
<button
  class="button button-primary"
/>
`;

exports[`<Button /> tests should render the button with some text 1`] = `
<div>
  <button
    class="button button-primary"
  >
    I am a button, I can be clicked
  </button>
</div>
`;
